### DISCUSSION TITLE
Questions and suggestions

### DISCUSSION THREAD

**Discussion by Vicen04** [Vicen04/Dw1DataAndPatches#16] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Welcome others and are open-minded. 
  * Keep everything civil.
  * Don't spam.

I'll try to check as much as possible and reply to most of the questions here.
The suggestions may or may not be implemented in the future, keep in mind that each one has its own ideals for the game.

**Comment by vlad8599** [comment]
Hello, great work with all this hacks, reworks and restoring cut content, thank you a lot. Can we have back in game Kaminarimon(I heard that it was a old name for Thunderballmon) and\or WereGarurumon? From what I learned, they both are in game code, ready to use, but dont have own models that was added in Card Battle.

**Reply by Vicen04** [reply]
Kaminarimon (which may or may not be related to Thunderballmon) has absolutely no data other than the name in that unused evolution item. The item has no ID set, so I literally have no idea what it was suppossed to be, but the partner slots are full and there was never any slot for that digimon in the released game.

While it is true that Weregarurumon have SOME data that can be used, a lot of it is incomplete and/or missing, it is not ready to be used at all. It does not have a finisher (Glacial Blast is not only broken, it is just a copy of Mail Storme).

The models from Card Battle would need to be reworked to make them compatible with World 1 and even with that, then the animations would need to be created from scratch...

**Reply by vlad8599** [reply]
I see. Sorry to hear that.

**Reply by Isshin-Ash1na** [reply]
actually Kaminarimon is not thunderballmon, rumors say it would be this one:
![816px-Kaminarimon_V_Jump_01.1999.jpeg](https://github.com/user-attachments/assets/c6f234f8-676b-4d8a-892b-d7989c4f6e60)

**Comment by mrchub12** [comment]
Hey! I didn't see a text file to follow for evolution, where should I be looking for that?

**Reply by Vicen04** [reply]
The txt README already explains how to get the new evolutions if you downloaded the RAR.
Otherwise, the installer has links to various spreadsheets that have all the data.

**Reply by mrchub12** [reply]
> The txt README already explains how to get the new evolutions if you downloaded the RAR.
> 
> Otherwise, the installer has links to various spreadsheets that have all the data.

I am referring to just the normal
evolutions, I am under the impression these mods follow different requirements for evolution. Was that only the older mods?

**Reply by Vicen04** [reply]
No, the Vice hack never changed any evolution requirements except for Gabumon -> Panjyamon in exchange of Gabumon -> Ogremon in the versions before 2.0 or with the "restore Panjyamon" patch.

**Comment by Naviel08** [comment]
Hi Vice, It's possible to add more maps to the game? I've been working on some special maps variations for some of the new events/zones on your patch, like the new boss battles at the spore area or the events before the weregarurumon battle. I would like to share my progress with you but first I wanted to ask if it is possible to add maps that are not in the vanilla game

**Reply by Vicen04** [reply]
It is possible... but it would require quite a bit of work... 
I have mentioned this before, but I want to take a break from this project.
I'm not the only person who is capable of modding the game, so you could ask for help in the digimon modding community discord and see if they can help to either add this so you could have a modified version of my hack or to make your own hack...

**Reply by Naviel08** [reply]
Oh I see... It's ok bro, you did an excelent work and made our dreams come true with that playable weregarurumon!! Anyways, I will leave to you what what I did in case that some day in the future you back to work on DW1 and if you like my idea you could add it on a future patch. Thank you for your efforts to improve the game! Hope you could check the files and the .txt with a explicaction of my ideas! :D
[Digimon-World-1-New-Maps.zip](https://github.com/user-attachments/files/17822539/Digimon-World-1-New-Maps.zip)

**Reply by Vicen04** [reply]
I'll reply to each of the points:
1. This one would be the easy to implement, but I'm not sure if it would be better to have a "special warp" there (or different colored, to show it is special).
2.  While the idea is not bad... This one is pretty difficult to implement...
3.  The sprite for this one can be changed without any issue, my idea was to actually make something that looks like a fusion of the Mt. Infinity/Back Dimension and a normal map, like it is a corrupted area.
4. It is a way to keep the item as a secret and to also give a hint as to what it could be... since you know, it can be easy to relate that item to what it may do...

**Reply by gabriel-iturra** [reply]
> Oh I see... It's ok bro, you did an excelent work and made our dreams come true with that playable weregarurumon!! Anyways, I will leave to you what what I did in case that some day in the future you back to work on DW1 and if you like my idea you could add it on a future patch. Thank you for your efforts to improve the game! Hope you could check the files and the .txt with a explicaction of my ideas! :D [Digimon-World-1-New-Maps.zip](https://github.com/user-attachments/files/17822539/Digimon-World-1-New-Maps.zip)

Apologize for being off-topic: just wanted to comment that your maps are fantastic. 
I love how the style and feel of the game's existing maps is kept overall.

**Comment by JhonyXY** [comment]
Como faço para entrar no discord? percebi que vocês mapearam boa parte do código do digimon world então vou fazer um programa que lê o código mapeado para que seja mais simples de fazer modificações no jogo e outras pessoas consigam fazer mods para esse jogo de forma simples sem entender de HxH.

**Reply by Vicen04** [reply]
If the translator is not wrong... you are looking for the digimon modding community discord link?

SydMontague has it on its Github, in the contact part of the DSCSTools repository.

**Comment by Lucas-RDV** [comment]
is it possible to change setting mid game? the tech boost i got from spit fire fells kinda too strong in the early game so i wanted to remove it or increase the dificulty but dont want to start a new save. the idea is good but i fell like the boost is too big to justify using anyother techs once you learn the boosted one

**Reply by Vicen04** [reply]
Yes, you can patch it again and use the exact same save. Just make sure you use a clean ROM when you're patching.

**Comment by thiviscar** [comment]
Hello, I've seen what you done with Poyomon being able to say which evolution your digimon is going to turn into. I was thinking if it was possible for Poyomon (or some other NPC) to also tell you how much time your digimon has until they evolve and, in case of Champions, how much time they still have to live if you don't evolve them.

**Reply by Vicen04** [reply]
It is possible... but I would need to find room for both the code and the dialogue... Which would be very complicated.

I cannot increase the Poyomon dialogue anymore as of now.

**Comment by javiermendezaguilar** [comment]
![digi](https://github.com/user-attachments/assets/7a34fc1e-45b5-478e-b07a-b9ccabf479de)

It would be interesting to replace the title screen with this image from the physical Japanese game cover. I have a friend who is an experienced illustrator, who says he could restore and repurpose this image to create a nice title screen for the game. If you are interested let me know and I will have my friend restore the image to suit your needs.

**Reply by Vicen04** [reply]
I would rather not do that, I'm pretty sure that actually can cause a copyright issue later on... It would also need to be in a format the game can handle... Which luckily is only TIM and not one of the other formats the game has...

**Reply by javiermendezaguilar** [reply]
What limitations do we have for modding? I.e. when can copyright be an issue? What is allowed and what is not allowed?

**Reply by Vicen04** [reply]
It is more about sharing those, you cannot share images/videos that are consedered a property without having any issues... it is fine if you change/add stuff that is your own work.

**Comment by javiermendezaguilar** [comment]
It would be interesting to have your hack, when you start the game, give you the option to select the language. And you can have your hack in Spanish, English, Japanese, or as many languages ​​as possible. That would bring the game community together, having a single game for everyone, multilingual.

**Reply by Vicen04** [reply]
That would require a huge amount of work, it is not possible to add another language to the USA version as of now... It just lacks characters and it doesn't have any room for more.

This is a visual representation of the full font:
![DW1USAFont](https://github.com/user-attachments/assets/71b53bb9-0bef-42a7-8f05-da6aaf848ab0)

**Reply by Bambusbomber** [reply]
Is it Possible for German? because german use the same characters.
For the original game there is an german patch so if it is possible it would be very nice to make the vice project work with german too.
my friend is an really big digimon world fan but he has many problems with english so it would be an nice present for him

**Reply by Vicen04** [reply]
The USA version can be translated to German (and I'm sure it would lack characters)... But the patch cannot be moved to the PAL version, the USA and each PAL versions are, in code terms, very different games. 

This is the font for the PAL versions:
<img width="276" height="145" alt="image" src="https://github.com/user-attachments/assets/cb8cb0a8-d089-4fcc-937c-d54b3b82a17b" />




I would need to start basically from 0 to do it for the German PAL (which is also a language I don't speak...) and that would only work for the German PAL and none of the other PAL.

**Reply by Bambusbomber** [reply]
Ah Okay.
Hmmmm sorry me englisch ist not perfect so when i understand you right then we can patch the German Patch on the NTSC Version?
I dont need an PAL Version on an Emulator an NTSC Version is completle fine.
Do you think it will work out of the box when i take an NTSC Version Patch VIce on it and then patch the German Language or need it do something else?

Im really new to play with modding roms

**Reply by Vicen04** [reply]
What I'm saying there is:
- If someone takes the USA version and translates it into German... you will be able to play it on German. 
As far as I'm aware, no one has done that...

**Comment by javiermendezaguilar** [comment]
Would it be possible to move the clock a bit more towards the top corner? Also, remove the happiness and discipline bars. Then, create a new HUD that includes absolutely all the alterable values in the game, such as fatigue, happiness, discipline, errors, etc., etc. To make the HUD appear on the screen, you press a button, and all the values are displayed. Then, you press that same button again, and the HUD becomes invisible.
![digi](https://github.com/user-attachments/assets/f7c537ba-c8d3-48b8-993d-6fe4fb254b82)

**Reply by Vicen04** [reply]
That would require new images, a lot of new code and a new font... since the current font is too big to even make that possible. You also can see all of that data by just opening the stats screen as of now, it may be better since it also pauses time... rather than wasting some of the lifespan just to try to read what's on the screen.

I'm also not sure what's the idea behind the food image... Is it the energy or something like that?

**Reply by javiermendezaguilar** [reply]
I can deliver new high-quality pixel art icons and a new font with the necessary size. You can always have those values displayed on the screen, and if at any point they bother you, you can press a button to hide them.

It's true that you added battles, errors, and fatigue, but values like remaining life hours are very important, and you didn't include them. The time left until going to sleep is also interesting, as well as a value that tells you when your Digimon will need to go to the bathroom. Additionally, food is energy, so knowing when your Digimon will need to eat again can help you anticipate whether a mushroom will be enough or if you should give more food. Displaying this value will help you better manage weight.

This approach is designed with the general public in mind. I understand that for experienced players or speedrunners, the approach might be different, but this modding is aimed at making the gaming experience more accessible for the general audience.

**Reply by Vicen04** [reply]
I'm going to be honest here:
It is a huge amount of work: you need room to add the code, add the images and the font. 
The digimon world font is not an image, it uses an special type of encoding which then is translated into "colored" and "transparent" pixels. To create a smaller font you need to write code to translate a smaller encoding.
There is also a problem with the amount of stuff you can render at a time, I'm not sure how the game will react if values that are constantly updated and drawn appear there... in some ocassions there is a chance the game will crash...

**Comment by javiermendezaguilar** [comment]
hi im majinwolf. Are you planning to add a marker that indicates how many hours your digimon has left to live? It would also be good if you added another marker that indicates how much time your digimon has left to evolve.

**Reply by Vicen04** [reply]
No, I'm not planning to add those.

**Comment by javiermendezaguilar** [comment]
I also thought it would be interesting when the player reaches a certain tamer level. to add some things like a non-perishable portable potty, that is, one that appears in the menu, just like the fishing rod does and when the digimon wants to go to the bathroom, press the portable potty icon and the digimon defecates. the bathroom, that would make that improve discipline and happiness, but nevertheless so that the conventional batteries do not become obsolete, it will make them increase happiness and double the discipline of the portable potty. I would also make it so that when the player increases his tamer level, he gets more money, and with another tamer level increase, the digimon would drop more money (it would eliminate the infinite money bug of mojyamon). In short, it makes better use of the tamer level mechanics so that the player feels an incentive to want to improve it.

**Comment by LinconEduardo** [comment]
in your changelog.

> [...] An item that used to be food, has been changed and now is an evolution item, another has been changed to give the same effects as the earlier item.

can yo say what itens you changed?

Another question, did you add any specific items to reduce the lifespan or reset it?

**Reply by Vicen04** [reply]
Late reply, but check the spreadsheet:
https://docs.google.com/spreadsheets/d/1lG3aLJsLiCwcZXo5-OS18o21GngTVuyAiKA0liV_kpM/edit?usp=sharing

**Reply by LinconEduardo** [reply]
I see, the lifespan reset is with any steak or only with the one sell at vending machine at Dragon Eye Lake?

maybe i misinterpreted, should be an item that increased lifespan, killing the digimon... i tested the digimon  can eat the golden acorn, can u make this change?

**Reply by Vicen04** [reply]
It only works with the "helpful items" patch, otherwise is just a normal steak.

I don't understant the second question, but if you're asking for a way to have infinite lifespan, digisieabass/ chain melon and the "better raise" patch is a way.

**Reply by LinconEduardo** [reply]
now i understand, the steak killed my digimon making a digitama... thats what i wanted, i wast thinking this ill reset his live to age 0 ^_^

there is a room to make de gold acorn another useful item, since the digimon can eat it anytime🤔 

ill be back with some feedback about the other useful items...

thank you for you awesome job!!

**Reply by Vicen04** [reply]
The Golden Acorns are just food, just like the Steak but with less room for custom code...

**Comment by RyuuzakiL5** [comment]
Ok, my question is something that I have tried hacking before but with no success:

Is it possible to hack the game and let Angemon learns "Thuder Justice"? I'm not asking you to make a patch, but if you can help me telling what are the offsets I need to change, I would be really happy.

He is my favorite Digimon, but I don't like to play with him because I hate most of his attacks. Devimon, on the other size (my second favorite) has two of my favorite attacks wich are TJ and Ice Statue.

**Reply by Vicen04** [reply]
The wiki has a little explanation and a tutorial to do that stuff... but it only works with the USA version.
It is located in the [Other data section](https://github.com/Vicen04/Dw1DataAndPatches/wiki/Other-data)
The offsets are in the "Other data" folder too

It already has Thunder Justice in the hack.

**Reply by RyuuzakiL5** [reply]
Oh tahnk you, you're my hero!

**Comment by LinconEduardo** [comment]
using the "real MetalGreymon patch" only dont apply in the arena lobby MetalGreymon

![Untitled](https://github.com/user-attachments/assets/cde4d676-9235-4256-92da-ed76b2be0849)

**Reply by Vicen04** [reply]
Because that is a different MetalGreymon model, it only applies to the playable one... I even changed its name, so it is a bit different...

**Reply by LinconEduardo** [reply]
the change is "MetalGreymon?" ?

I found strange because i faced him in the arena with the right colors... u cant swap this model colors?

**Reply by Vicen04** [reply]
The real MetalGreymon is not just a "color swap", it also has changes in the model. I just don't want to change all three models in the game.

The arena uses the playable models, not the NPC ones.

**Comment by LinconEduardo** [comment]
about the usefull itens patch...

1 - Digipine, why not put it to sell in the gym i only found seeling in shogungekomon?
i mean the shogungekomon seeling (in my view) should be only rare floops, chips and evo itens... why sell food itens?

<details>
<summary>show details...</summary>
<img width="1320" height="540" alt="image" src="https://github.com/user-attachments/assets/d107e478-9b4a-47b4-b3b8-6db6f62f2cab" />
</details>

2 - Still talking about Digipine, the efects of the item is lasting forever (even when my digimon become digitama)... in my test the only way too stop the buff is eating another training item, i dont have tested other training itens yet to see if have the same problem...

3 - Still talking about Digipine and other items, why not make a better description about some itens like:

Digipine - Boosts ALL training effects.
Hawk radish - Increase weight, boosts training effect HP, DEF and SPD.
Supercarrot - Lowers weight, boosts training effect MP, OFF and BRN.
Gold Acorn - Increase weight, can be sold for a high price. (A few days ago I discovered that it was edible 😆)

(i have way more suggestions, if u want i cant list all...)

4 - Gekomon seeling moldy meat (?), why not put to sell digiseabass?

<details>
<summary>show details...</summary>
<img width="1320" height="540" alt="image" src="https://github.com/user-attachments/assets/3b830824-c2b2-465d-8053-ec73332cc987" />
</details>

**Reply by Vicen04** [reply]
1. This question has been asked before, I'm not going to add it to the gym because is too OP. Shogun sells rare items and that's all, it is up to the opinions of anyone. The digipine can only be found on the Misty Trees which is next to the Swamp and it is very rare.
2. The effects don't last forever, but the timer only goes down while the clock is moving naturally (while you walk or wait for stuff), if you wait doing nothing for a while or just spend time moving around... The effect will disappear at some point. It is also not reset on evolution or death, the devs just forgot that I guess... I'm not planning to change that as of now.
3.  There is no room for more text. Pretty much all the food gives weight and as I mentioned, the gold Acorn is food... It is called "Acorn".
4. Gekomon sells moldy meat just for ease of access to it. Selling digiseabass would be too OP because those give extra lifespan and heal all your HP/MP

**Comment by Miraak88** [comment]
A question. As a player who had the game back in the day but didn't go beyond having a Leomon and exploring the area around the city, what installation setup would you recommend?

**Reply by Vicen04** [reply]
If you have the latest version, you should have this setting at the top left corner:
<img width="1129" height="306" alt="image" src="https://github.com/user-attachments/assets/f0109e18-754a-443c-a9c5-c57732edc144" />


Choose the "Recommended" option and see if you would prefer the normal recommended settings or to play as vanilla as possible.
<img width="601" height="598" alt="image" src="https://github.com/user-attachments/assets/7b1c90e4-3866-4a16-8f8c-22fac18d15a0" />

All of the bug fixes come with the hack by default, so don't worry about that.

**Comment by Miraak88** [comment]
I've already finished the game with all the Digimon in the city. Great job, thank you! One suggestion: the items used to evolve the extra Digimon shouldn't be so easy to obtain. The Moon Mirror is fine since it's in the secret area, but the Metal Banana and Giga Hand shouldn't be drops from Etemon and MegaDramon. It would be better to add MetalEtemon and GigaDramon as hidden bosses and have them drop those items instead.

**Reply by Vicen04** [reply]
The first ones are easy to obtain so the player can learn that the items exist... then getting more is complicated.
MetalEtemon and Gigadramon are already hidden bosses, but it is up to your luck to find them...

**Reply by Miraak88** [reply]
I've sent you a message via contact form. Let me know what you think when you can, please.

**Comment by Miraak88** [comment]
<img width="1023" height="706" alt="382196144-757fa696-22b2-41ad-9893-766c0963885e" src="https://github.com/user-attachments/assets/b616d6a4-24d2-4af4-9803-af9e472b651e" />
<img width="1023" height="706" alt="h" src="https://github.com/user-attachments/assets/2a107d37-e0ec-4a9e-aef5-bfc330686b91" />
Some ideas for a possible updated evolution chart.

**Reply by Vicen04** [reply]
Out of those, the first one is the most likely to happen without the split with the orange... but I would need to also program the way the selection works... which is a huge issue.

**Comment by Miraak88** [comment]


**Reply by Vicen04** [reply]
I'll keep the original for now... I would rather fix all of those when a proper retranslation is done... and fix stuff like  "happy mushroom" actually being named "lucky mushroom" in the original...

**Comment by Miraak88** [comment]
One question @Vicen04 which tools I need to use in order to edit the game text?

**Reply by Vicen04** [reply]
I use Windhex and a table to be able to read the text... but you have to either learn how the game script works to be able to expand the text or work with the text limit...

I'll attach the table, it only works with the USA version.

[Digimon table.zip](https://github.com/user-attachments/files/21980800/Digimon.table.zip)

**Reply by Miraak88** [reply]
Is there any tool to compare the changes made in WindHex? I mean with the text readable, like WinMerge. I tried the latter since it can compare in hexadecimal, but it doesn’t support loading a table.

Thanks for all the help.

EDIT: Found a way. Dump text with WindHex, both the base DG file, and the edited one by me. Then compare with WinMerge.

**Comment by Miraak88** [comment]
I'm using your mod as a base to implement some fixes. I've kept the vanilla mode, Myotismon, and the Quick Battle Text, so I've fixed the vanilla vending machines and added your restaurant fix. Then, I went through all the text and replaced all written-out numbers with digits (just money and stats), along with some typo corrections and adjustments to improve how everything fits and looks in the text boxes.

There's something that's caught my attention. In the chart, the Digimon groups are organized by type, except for the Ultimates. Is there a way to reorder them? I suppose it shouldn't be too hard if I can find the right addresses... I looked at your patch that enables the extra Digimon in the chart, but it seems you're using different addresses for that, so I'm kind of in the dark here... Where should I look?

**Reply by Vicen04** [reply]
Fixed the vending machines?

Since you only want to change the location of those... You would need to swap the position and the sprites used for each square... 

Luckily, those are read for a table rather than be hard coded (I cannot say the same about the input, if you move a square just 1 pixel off, you won't see anything)

How it works (from what I remember):
A total of 8 bytes.
1 short for the position X
1 short for the position Y
1 byte for the position X in the texture 
1 byte for the position Y in the texture
1 byte for the CLUT ID
1 empty byte

The start... I don't have it right now, but it should be before 0x14D64E24... Considering it is about 65 Digimon... Maybe about 520 bytes before that...

**Reply by Miraak88** [reply]
> Fixed the vending machines?

Yep.  Gear Savanna machine bonus Double Floppy gives a Super Speed Disk instead if you have full inventory, and Anc. Dino Region machine gives Moldy Meat instead Steak and it doesn't even gives you the MP Floppy. These are very simple stuff for you to implement if you want.

**Comment by Miraak88** [comment]
Alright, I finished reorganizing the chart. In the end, I realized that the original order of the Ultimates makes more sense than it seems, so I only moved things around a little bit. That said, I left the 3 Megas in the Champion section, so now there are only Ultimates in the orange zone.

Here is the patch: [dwv_custom.zip](https://github.com/user-attachments/files/22083682/dwv_custom.zip)

A Vice patched ISO is needed, with this config:
- Myotismon Vice (Digimon)
- Original ultimate lifespan (Misc.)
- Remove Tech Boost (Misc.)
- Original Type Balance (Misc.)
- Remove evo info (Misc.)
- Quick Battle text (Misc.)
- Rare Spawns: 3% (Useful)

Changes:
- Fixed some typos/inconsistencies and incorrect info
- Changed to digits all writen numbers referred to money and stats
- Changed some item names/descriptions for better consistency/fit the menu boxes (Happymushrm > Lucky Mshrm)
- Changed (slightly) some music and places names to better fit the menu boxes
- Slightly reordered chart (orange zone has all Ultimates now)
- Fixed Gear Savanna and Anc. Dino Region vending machines (bonus Double Flop, and MP Floppy)
- Restaurant infinite food glitch fixed (Vice04)

Use whatever you need @Vicen04

**Reply by Vicen04** [reply]
I do remember there was a reason that made sense for the Ultimates order... but I forgot about it...

Just a couple of things to note:

I imagine you just saw the "number 2" when you checked the chance offset and assumed that was the actual chance... but it doesn't work like that, it is actually a "trigger if the value is equal or lower than 2", which would give a chance to trigger if it is 0, 1 or 2... a total of 3 numbers our of a possible 100 (max is 99). 

As for the Ancient Dino Region vending machine... that one is meant to sell "moldy meat" not "Steak", the "Steak" text is the mistake... otherwise the player would be completely unable to adquire moldy meat anywhere (and you could also sequence break Myotismon's mansion 1st visit). 


As for the changes... I probably won't use them... I just like the typos, that's all. As I said, if I ever fix it, I would do a complete retranslation to be accurate... there is a lot of stuff that is wrong and a lot of text that gives you useful info missing...

But I'm kinda happy the info is helping the development of a customized version. If you're interested in changing even more, I would recommend to head to the Modding community discord, since it has more info about the game itself (the Github is mostly within the hack), and people which can help.

**Reply by Miraak88** [reply]
> I do remember there was a reason that made sense for the Ultimates order... but I forgot about it...
> 
> Just a couple of things to note:
> 
> I imagine you just saw the "number 2" when you checked the chance offset and assumed that was the actual chance... but it doesn't work like that, it is actually a "trigger if the value is equal or lower than 2", which would give a chance to trigger if it is 0, 1 or 2... a total of 3 numbers our of a possible 100 (max is 99).
> 
> As for the Ancient Dino Region vending machine... that one is meant to sell "moldy meat" not "Steak", the "Steak" text is the mistake... otherwise the player would be completely unable to adquire moldy meat anywhere (and you could also sequence break Myotismon's mansion 1st visit).
> 
> As for the changes... I probably won't use them... I just like the typos, that's all. As I said, if I ever fix it, I would do a complete retranslation to be accurate... there is a lot of stuff that is wrong and a lot of text that gives you useful info missing...
> 
> But I'm kinda happy the info is helping the development of a customized version. If you're interested in changing even more, I would recommend to head to the Modding community discord, since it has more info about the game itself (the Github is mostly within the hack), and people which can help.

Fixed... Thanks!!

**Comment by Miraak88** [comment]
Since there's already a decompiler for DG files, I've started working on my own hack. Honestly, I just want something relatively simple, but I don’t have much knowledge, so I’m examining everything I can and using your bugfixes as base. For now, I have revised all the game’s text and fixed quite a few bugs in the scripts.

Right now, I’m trying to implement fixes for the HUD and the status menu. I see this in your code: 
```
800a9414 4b 00 06 24 li a2,0x4b 
800a9450 4b 00 06 24 li a2,0x4b 
800a97a4 04 00 06 24 li a2,0x4 
800a97a4 02 00 06 24 li a2,0x2
```
I understand the first two are the positions of the happiness and discipline bars, but I don’t know what the last two do, I don’t see any obvious changes. I’m also looking into how to implement your fixes for the stats screen, the virus bar, the happiness stat, etc, and the Gigadramon spite fix for the tournaments.

After this I want to replace Panjyamon for WereGarurumon, fix Gigadramon, change acouple of techs, add the evo items to the Merit shop and add the extra digimons to the respective tournaments. Maybe I'll try a R-MetalGreymon > Gigadramon replacement if I feel motivated enough.

It would be quite grateful if you can poitn me in the right direction.

**Reply by Vicen04** [reply]
If I'm not wrong... The first two are the length of the bar that has to render...
The second is a comparison, the first one being the original and the second one the changed version, that is a fix for the "overflow" in the discipline bar when is maxed.

Except for the tournament Digimon, the Evo items you want to add and the technique changes (kinda)... everything else requires knowledge about the code and being able to change that... Which is far from simple... 

For the tournaments: you can set which Digimon can participate/appear with the script. It is labeled as unknown 6B I believe... Right after the tournament descriptions...

The merit shop is pretty simple, you only have to look for the name of the item to find the item data, I think one of the spreadsheets from the modding discord has info about that? Otherwise is just setting a merit price, that should make those appear at the merit shop.

There is more or less an explanation about how to change techniques in the wiki here, the other data part... I'm not sure how helpful it will be, but you can take your time there.


As for the rest...

I don't know how much knowledge you have about disassembly... The program I use to check the code is Ghidra (I still have to manually setup the data, so it is readable and makes sense), it is mostly reverse engineering the game and trying stuff... 

I have not finished writing the changes in 2.1 because there are a lot and I'm also pretty busy with everything...

As mentioned, the modding discord can help too...

**Comment by Miraak88** [comment]
Thanks. One more thing, were I can find the stat screen and savegame screen changes?

**Reply by Vicen04** [reply]
> I have not finished writing the changes in 2.1 because there are a lot and I'm also pretty busy with everything...

**Comment by uwuaru** [comment]
First off i just want to say amazing work
just adding WereGarurumon is more than enough to make me want to replay the game again. he is one of my favorite characters
if there is a way i can convince you to change they myotismon patch or add a option to it that would be cool.
currently you have Myotismon replacing Machinedramon
if it could replace MetalEtemon instead you can remove the redundancy of the game having two etemon characters in the same way you did with the two Leomon characters by using WereGarurumon instead of Panjyamon

**Reply by Vicen04** [reply]
The way the Machinedramon/Myotismon addition works is special... I'm not adding new models or anything else, just calling an "improved" NPC version of it... If I want to replace either Gigadramon or MetalEtemon... I would need to replace the models and also... add animations for Myotismon.  

There is also another reason why I would not do that, but I'm not going to go into details now...

**Reply by uwuaru** [reply]
oh ok sounds good. you got Weregarurumon working which is already more than i thought would ever happen

**Comment by h4tter** [comment]
Excellent work with all the hacks and mods!
I thought I saw something about a QoL mod where the Mojyamon that joins the city gets replaced in Freezeland afterwards, so you don't lose out on their trades. But I can no longer find anything about it in the patcher, so perhaps I imagined it. Is there something like that in Vice 2.2? And if not, is that something that could be modded?

**Reply by Vicen04** [reply]
I'm pretty sure there was not option like that at all... While the idea is not impossible, you usually just recruit the Mojyamon that gives you bad trades (the one at the right side in the map with 2 Mojyamons), since that one is usually useless.

**Reply by h4tter** [reply]
Ah, thank you, I think I had seen "Restore Panjamon", confused Panjamon with Mojyamon, and assumed that patch "restored" the recruited Mojyamon. But you're right, I'd forgotten how bad the right Mojyamon's trades are, it's pretty much useless.

Is the purpose of the Steak from the Useful Items that it means you can move on to evolving other Digimon quicker, without having to risk losing tamer levels by losing battles or using devil chips?

I feel it would also be useful to be able to skip days, to get the 10 year medal easier, or shift to a better position in the arena calendar. Again, having looked at all the spreadsheets and the wiki, I feel like I saw something about skipping time, but now I can't seem to find it. Is there anything in the Vice mod for skipping time, or have I misremembered that too?

Finally, is there information on the Better Vending Machines patch, that I just haven't been able to find? I couldn't find it in any of the spreadsheets, notably the optional patches sheet where I'd expect those details to be.

Thanks again for your help!

**Reply by Vicen04** [reply]
1. Yes the Steak is mostly to make the Digimon pass away and fill the chart faster.
2. You can skip a full year by an in game event. I'm not sure if I left the details, it may either be on a spreadsheet or a video. It has to do with the special Sukamon.
3. The better vending machines just makes the vending machines be actually useful... There are only 4 in the entire game and the patch only modified 3 of those. It felt like there was too little for me to make a page about it. If you're curious the changes are:

-  The vending machine at the Tropical forest will offer better healing.
- The vending machine at the Ancient Dino region will give better items
- The vending machine in Gear Savanna will give you better and cheap healing items.

**Comment by jrd656** [comment]
Loving the mod. Thank you for the crazy hours you must have put in.

I'm trying to find the details of what guardramon upgrade will evolve which digimon. I know the og was mamemon. I believe metal greymon is one of the new ones. Is this info available anywhere? I can't find it in the spreadsheets or changelog, but I feel like I've read it somewhere already?

Also, I think I may have installed patch 2.0. If I install a newer patch, will I be able to still use the old save?

**Reply by Vicen04** [reply]
The main spreadsheet has the data in the "Factorial town" page:
[Spreadsheet Vice 2.1/2.2](https://docs.google.com/spreadsheets/d/1lG3aLJsLiCwcZXo5-OS18o21GngTVuyAiKA0liV_kpM/edit?usp=sharing)

You can play a later version with the same save file, just make sure your current partner is not Machinedramon/Myotismon or any Digimon that can be changed with optional patches (or just make sure you have the same options for Digimon).

**Comment by EnemyRaptorC** [comment]
Thank you for all your work on this! This is a really incredible effort, and the result is the definitive Digimon World experience to me.
I have two difficulty option suggestions:

1. Option to change the Big HP Flop -> S.Def. Disk trade with Mojyamon to Super MP -> S.Def. Disk so that it's no longer possible to generate unlimited Bits by repeatedly doing this trade.
2. Option to change the item stack limit to something like 10~15. You can pretty easily spam items to get through anything so long as you have an Ultimate level Digimon. Imposing a severe limit on item stacks would be similar to the Fair Battles option without completely removing the purpose of items like Various and Protection.

I also have one maybe absurd pipe dream suggestion:
For Digimon that can be obtained, overlay icons on their card for which Digimon can Digivolve to them and which stats are in their requirements. Why do this with cards? Because you can look at a card of a Digimon before you've raised it, and it gives a reason to collect the cards beyond merit points and the medal for them. I think the icons from the existing Digivolution chart and status page would work fine. EDIT: This is a rough mock-up of what I mean:
<img width="1280" height="960" alt="Digimon Card Hint mock-up" src="https://github.com/user-attachments/assets/a9fa8963-1a0a-4744-ac85-23a5a4ba84a2" />

**Reply by Vicen04** [reply]
Both suggestions are stuff that the player can just... Choose not to do, there are even easier ways to make money depending on your options... And protection is kinda overpowered, you only need 1 per battle and it makes everything a lot easier...

As for the cards thing: I would rather not do that and ruin the cards art... I already had an idea about how to go for requirements, but I'm not even sure if I'll ever reach that point.

**Comment by Miraak88** [comment]
God! You’ve polished the menus spectacularly… There are still a few small details left, but it’s an incredible piece of work.
I’m considering switching back to your patch. You already know you can use everything I did in my patch without any issues. You can find it on Romhack Plaza. Inside the zip, all the modified files are included separately.
By the way, is it now possible to extract the ISO with your patch, or are you still working the same way?

Thanks for your work, seriously.

**Reply by Vicen04** [reply]
I'm still working in the same way, but there should be now less issues when you extract it... there is still a lot to do in that regard anyway...

**Comment by Viccxinfection** [comment]
Im sorry if this is already stated somewhere but couldnt find any info about it, but im trying to play with the "true hardcore" patch, i tried with nerf tech boost and effect boost and various options, but didnt find a way to change it. Im randomly getting 9999 dmg from enemies and my digis tech, it doesnt seems to be tech boost because its happening on various tech (IE magma bomb with agumon agumon) and i dont seem to find a way to play true hardcore without that 9999 dmg rng, is it a feature or is it bugged? thanks in advance

**Reply by Vicen04** [reply]
Okay, it seems that exclusively in the "true hardcore" patch some leftover code was changing the last damage calculation: the one that makes sure the damage doesn't go over 9999... 
This made it... rather than check if the value is 10K or more... check if it is 300 or more and then if the value is above that set it to 9999 then...
I just fixed it, but I need some time to compile and upload the installers again... It should be done soon.
Thanks for the info... I'm pretty sure I did check for errors (specially since I changed all of the damage code to prepare it for part 3) so I'm not completely sure when this one appeared...

**Reply by Viccxinfection** [reply]
Thanks for the quick reply, since no one talked about it i though it was a problem on my end, glad it could be resolved!
